### PR TITLE
fix: skip the confirm save modal when the user do not have permission

### PR DESCRIPTION
**Pull request by wa0x6e**
### Summary

When navigating the space settings page, and leaving after browsing the different tab (strategies, delegation, etc...), it sometimes open a confirmation modal dialog, asking to confirm if we want to leave, due to unsaved changes.

This confirmation modal should be presented only to users with write permission (space controller and admins), and is irrelevant to other users.

### How to test

1. Go to a space settings page you don't have permissions, e.g https://snapshot-git-skip-save-modal-when-no-permission-snapshot.vercel.app/#/stgdao.eth/settings
2. Go to delegation tab
3. input something in the inputs
4. Click on some link to leave the space settings
5. before fix, it should open the confirm dialog, after fix, it just leave.
6. Repeat the same on a space you have admin permission
7. on leave, it should open the confirm dialog

### To-Do

- [ ]

### CODE REVIEW

**Comment by ChaituVR** [comment]
Running `yarn format:fix` is getting tedious, may be better to run it on commit hook?

**Comment by wa0x6e** [comment]
> Running `yarn format:fix` is getting tedious, may be better to run it on commit hook?

I think it was removed for speed issue.

**Review by ChaituVR — APPROVED** [review]
tAck

File: src/views/SpaceSettings.vue [hunk 1]
```diff
@@ -168,16 +168,16 @@ const {
   cancel: cancelDelete
 } = useConfirmDialog();
 
+const isViewOnly = computed(() => {
+  return !(isSpaceController.value || isSpaceAdmin.value);
+});
+
 onBeforeRouteLeave(async () => {
-  if (hasFormChanged.value) {
+  if (hasFormChanged.value && (isSpaceController.value || isSpaceAdmin.value)) {
```
Comment (samuveth) [inline src/views/SpaceSettings.vue:176]: Why did you move `isViewOnly` up and then not use it?

**Comment by samuveth** [comment]
@ChaituVR if your editor is setup properly you should never have to run `format`

**Comment by ChaituVR** [comment]
> @ChaituVR if your editor is setup properly with Prettier you should never have to run the format

Yea it works great with all repos, but something is missing in this repo 🤔

File: src/views/SpaceSettings.vue (hunk 1 above)
Comment (wa0x6e) [reply src/views/SpaceSettings.vue:176]: It was my intention to use it, but was not working for some reason

**Comment by wa0x6e** [comment]
> > @ChaituVR if your editor is setup properly with Prettier you should never have to run the format
> 
> Yea it works nicely with all repos, but something is missing in this repo 🤔

Correct, I rely 100% on code editor for formatting, that’s why I missed it

**Comment by samuveth** [comment]
@ChaituVR @wa0x6e are you sure prettier is your default formatter?

File: src/views/SpaceSettings.vue (hunk 1 above)
Comment (samuveth) [reply src/views/SpaceSettings.vue]: Pushed something. Please check @wa0x6e

**Comment by wa0x6e** [comment]
> @ChaituVR @wa0x6e are you sure prettier is your default formatter?

Ah, I was missing the `.value`, explain why it wasn't working

**Comment by samuveth** [comment]
> > @ChaituVR @wa0x6e are you sure prettier is your default formatter?
> 
> Ah, I was missing the `.value`, explain why it wasn't working

How is that related to prettier?

**Comment by wa0x6e** [comment]
> > > @ChaituVR @wa0x6e are you sure prettier is your default formatter?
> > 
> > 
> > Ah, I was missing the `.value`, explain why it wasn't working
> 
> How is that related to prettier?

Wrong quote, was referring to https://github.com/snapshot-labs/snapshot/pull/4343#discussion_r1389008013
